lib/amqp.js

```diff
@@ -0,0 +1,172 @@
+var debug = require('debug')('sailor');
+var amqplib = require('amqplib');
+var cipher = require('./cipher.js');
+var Q = require('q');
+var _ = require('lodash');
+
+var AMQPConnection = function(settings) {
+
+    var self = this;
+
+    self.amqp = null;
+    self.subscribeChannel = null;
+    self.publishChannel = null;
+
+    self.connect = function connect(uri) {
+        return openConnection(uri)
+            .then(openSubscribeChannel)
+            .then(openPublishChannel)
+            .then(assertAllQueues);
+    };
+
+    function openConnection(uri) {
+        return amqplib.connect(uri).then(function(connection){
+            debug('Connected to AMQP on %s', uri);
+            self.amqp = connection;
+        });
+    }
+
+    function openSubscribeChannel() {
+        debug('Open subscribe channel');
+        return self.amqp.createChannel().then(function(channel) {
+            debug('Opened subscribe channel');
+            self.subscribeChannel = channel;
+        })
+    }
+
+    function openPublishChannel() {
+        debug('Open publish channel');
+        return self.amqp.createChannel().then(function(channel) {
+            debug('Opened publish channel');
+            self.publishChannel = channel;
+        });
+    }
+
+    function assertAllQueues() {
+        return Q.all([
+            assertQueue(self.subscribeChannel, settings.INCOMING_MESSAGES_QUEUE),
```

Comment:
It's OK, so sailor will only connect.. 
But now we have no that "entity" yet. 
Maybe I'll just make a script which creates queues (to use & test on localhost)?